### DISCUSSION TITLE
Welcome to simple-node-editor Discussions!

### DISCUSSION THREAD

**Discussion by bhowiebkr** [bhowiebkr/python-node-editor#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jimkring** [comment]
Hi @bhowiebkr. I was curious if you've used this in any applications.  I'm a [LabVIEW](https://ni.com/labview) programmer (and also use Python a lot, too), so I'm familiar with node-based programming. :)

**Reply by bhowiebkr** [reply]
Sofar nothing. I like the concept of nodes but never found a good use for them in my own code. My experience with nodes is in Autodesk Maya when I used to be a rigger. All kinds of messy node graphs. Many VFX tools are node based. Then in Unreal Engine which works great with nodes from blueprints to shaders. But I find it difficult to organize and mentally maintain a node graph of code in my head.  In the case of Unreal Engine once I get lost in the nodes I redo it all in C++ 

Sebastian Lague’s series on YouTube of Exploring how Computers Work is really interesting and I’ve thought about remaking that with this node editor so I can follow along. 

Alot of this python node editor code I wrote a long long time ago and it was used as a visualizer tool to show dependencies of assets and their relation to shots in a VFX pipeline. 

I’d like to turn this into a general python computing tool with nodes where you write your python code in a new node type and it communicates with connected nodes through Qt’s signals and slot framework. Have a concept of caching nodes if their state doesn’t change. 

I don’t know how people are using this code but I feel like it’s incomplete and I bet many people are downloading this code and having to implement the same common missing things: save/load graph, implement some kind of logic on the nodes so they do something.

**Reply by jimkring** [reply]
@bhowiebkr -- I like where your ideas! 👍
